mm2src/coins/lp_coins.rs

```diff
@@ -599,6 +594,8 @@ pub trait SwapOps {
 
     fn extract_secret(&self, secret_hash: &[u8], spend_tx: &[u8]) -> Result<Vec<u8>, String>;
 
+    fn check_all_inputs_signed_by_pub(&self, tx: &[u8], expected_pub: &[u8]) -> Result<bool, String>;
```

Comment:
`all_inputs` is UTXO specific, the name doesn't match other coins protocols. `check_tx_signed_by_pub` seems a better match for me.